backend/tests/controllers/auth.controller.test.ts

```diff
@@ -0,0 +1,154 @@
+import { Request, Response } from 'express';
+import { register, login } from '../../src/controllers/auth.controller';
+import { createUser } from '../../src/services/user.service';
+import { getUserTokens } from '../../src/services/auth.service';
+
+jest.mock('../../src/services/user.service', () => ({
+  createUser: jest.fn(),
+}));
+
+describe('register', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should create a new user and return 201', async () => {
+    const newUser = {
+      userName: 'testuser',
+      email: 'test@example.com',
+      password: 'password',
+    };
+    const userData = {
+      userName: 'testuser',
+      email: 'test@example.com',
+      password: 'password',
+    };
+    (createUser as jest.Mock).mockResolvedValueOnce(newUser);
+
+    const req = { body: userData } as Request;
+    const res = {
+      status: jest.fn().mockReturnThis(),
+      json: jest.fn(),
+    } as unknown as Response;
```

Comment:
Extract this to util functions. Do not repeat yourself (DRY Principle)
```suggestion
    /**
 * Creates a mock request object.
 * @param body The request body.
 * @returns Mock request object.
 */
export const createMockRequest = (body: any): Request => {
  return { body } as Request;
};

/**
 * Creates a mock response object.
 * @returns Mock response object.
 */
export const createMockResponse = (): Response => {
  const res: Response = {
    status: jest.fn().mockReturnThis(),
    json: jest.fn(),
    send: jest.fn(),
  } as unknown as Response;
  return res;
};
```